File: bus.java
import java.util.*;

class Car {
    private String carId;
    private String make;
    private String model;
    private double pricePerDay;
    private boolean available;

    public Car(String carId, String make, String model, double pricePerDay) {
        this.carId = carId;
        this.make = make;
        this.model = model;
        this.pricePerDay = pricePerDay;
        this.available = true;
    }

    public String getCarId() {
        return carId;
    }

    public String getMake() {
        return make;
    }

    public String getModel() {
        return model;
    }

    public double getPricePerDay() {
        return pricePerDay;
    }

    public boolean isAvailable() {
        return available;
    }

    public void setAvailable(boolean available) {
        this.available = available;
    }
}

class RentalSystem {
    private List<Car> cars;
    private Map<String, List<Car>> reservations;

    public RentalSystem() {
        cars = new ArrayList<>();
        reservations = new HashMap<>();
    }

    public void addCar(Car car) {
        cars.add(car);
    }

    public void removeCar(Car car) {
        cars.remove(car);
    }

    public void makeReservation(String customerId, Car car, int days) {
        if (!car.isAvailable()) {
            System.out.println("Car " + car.getCarId() + " is not available for the selected dates.");
            return;
        }

        if (!reservations.containsKey(customerId))
            reservations.put(customerId, new ArrayList<>());

        reservations.get(customerId).add(car);
        car.setAvailable(false);
        System.out.println("Reservation made successfully for Car " + car.getCarId() + " for " + days + " days.");
    }

    public void returnCar(String customerId, Car car) {
        if (!reservations.containsKey(customerId)) {
            System.out.println("No reservations found for customer " + customerId);
            return;
        }

        List<Car> customerReservations = reservations.get(customerId);
        if (!customerReservations.contains(car)) {
            System.out.println("Car " + car.getCarId() + " is not reserved for customer " + customerId);
            return;
        }

        customerReservations.remove(car);
        car.setAvailable(true);
        System.out.println("Car " + car.getCarId() + " returned successfully.");
    }

    public void generateInvoice(String customerId) {
        if (!reservations.containsKey(customerId)) {
            System.out.println("No reservations found for customer " + customerId);
            return;
        }

        List<Car> customerReservations = reservations.get(customerId);
        double totalCost = 0.0;

        System.out.println("\nInvoice for Customer " + customerId);
        System.out.println("--------------------------------------------");
        for (Car car : customerReservations) {
            double cost = car.getPricePerDay();
            totalCost += cost;
            System.out.println("Car ID: " + car.getCarId() + ", Make: " + car.getMake() + ", Model: " + car.getModel() + ", Cost per day: $" + cost);
        }
        System.out.println("--------------------------------------------");
        System.out.println("Total Cost: $" + totalCost);
    }
}

public class CarRentalSystem {
    public static void main(String[] args) {
        RentalSystem rentalSystem = new RentalSystem();

        Car car1 = new Car("C001", "Toyota", "Camry", 50.0);
        Car car2 = new Car("C002", "Honda", "Accord", 60.0);
        Car car3 = new Car("C003", "Ford", "Mustang", 100.0);

        rentalSystem.addCar(car1);
        rentalSystem.addCar(car2);
        rentalSystem.addCar(car3);

        rentalSystem.makeReservation("CUST001", car1, 5);
        rentalSystem.makeReservation("CUST001", car2, 3);
        rentalSystem.makeReservation("CUST002", car3, 7);

        rentalSystem.generateInvoice("CUST001");

        rentalSystem.returnCar("CUST001", car1);
        rentalSystem.returnCar("CUST002", car3);

        rentalSystem.generateInvoice("CUST001");
        rentalSystem.generateInvoice("CUST002");
    }
}

File: bus.py
class Car:
    def __init__(self, car_id, make, model, price_per_day):
        self.car_id = car_id
        self.make = make
        self.model = model
        self.price_per_day = price_per_day
        self.available = True

    def get_car_id(self):
        return self.car_id

    def get_make(self):
        return self.make

    def get_model(self):
        return self.model

    def get_price_per_day(self):
        return self.price_per_day

    def is_available(self):
        return self.available

    def set_available(self, available):
        self.available = available


class RentalSystem:
    def __init__(self):
        self.cars = []
        self.reservations = {}

    def add_car(self, car):
        self.cars.append(car)

    def remove_car(self, car):
        self.cars.remove(car)

    def make_reservation(self, customer_id, car, days):
        if not car.is_available():
            print(f"Car {car.get_car_id()} is not available for the selected dates.")
            return

        if customer_id not in self.reservations:
            self.reservations[customer_id] = []

        self.reservations[customer_id].append(car)
        car.set_available(False)
        print(f"Reservation made successfully for Car {car.get_car_id()} for {days} days.")

    def return_car(self, customer_id, car):
        if customer_id not in self.reservations:
            print(f"No reservations found for customer {customer_id}")
            return

        customer_reservations = self.reservations[customer_id]
        if car not in customer_reservations:
            print(f"Car {car.get_car_id()} is not reserved for customer {customer_id}")
            return

        customer_reservations.remove(car)
        car.set_available(True)
        print(f"Car {car.get_car_id()} returned successfully.")

    def generate_invoice(self, customer_id):
        if customer_id not in self.reservations:
            print(f"No reservations found for customer {customer_id}")
            return

        customer_reservations = self.reservations[customer_id]
        total_cost = 0.0

        print("\nInvoice for Customer", customer_id)
        print("--------------------------------------------")
        for car in customer_reservations:
            cost = car.get_price_per_day()
            total_cost += cost
            print(
                f"Car ID: {car.get_car_id()}, Make: {car.get_make()}, Model: {car.get_model()}, Cost per day: ${cost}"
            )
        print("--------------------------------------------")
        print("Total Cost: $", total_cost)


def main():
    rental_system = RentalSystem()

    car1 = Car("C001", "Toyota", "Camry", 50.0)
    car2 = Car("C002", "Honda", "Accord", 60.0)
    car3 = Car("C003", "Ford", "Mustang", 100.0)

    rental_system.add_car(car1)
    rental_system.add_car(car2)
    rental_system.add_car(car3)

    rental_system.make_reservation("CUST001", car1, 5)
    rental_system.make_reservation("CUST001", car2, 3)
    rental_system.make_reservation("CUST002", car3, 7)

    rental_system.generate_invoice("CUST001")

    rental_system.return_car("CUST001", car1)
    rental_system.return_car("CUST002", car3)

    rental_system.generate_invoice("CUST001")
    rental_system.generate_invoice("CUST002")


if __name__ == "__main__":
    main()